packages/allow-scripts/src/setup.js

```diff
@@ -2,73 +2,133 @@ const { existsSync,
+  const rcPath = addInstallParentDir(file)
+  process._rawDebug({
+    cwd: process.cwd(),
+    rcPath,
+    ex: existsSync(rcPath)
+  })
+  if (!existsSync(rcPath)) {
+    return false
   }
-
   const rcFileContents = readFileSync(rcPath, 'utf8')
-  if (rcFileContents.includes(entry)) {
+  process._rawDebug({ isentry: rcFileContents.includes(entry)})
+  return rcFileContents.includes(entry)
+}
+
+function writeRcFileContent({file, entry}){
+  const rcPath = addInstallParentDir(file)
+
+  if (isEntryPresent(entry, file)) {
     console.log(`@lavamoat/allow-scripts: file ${rcPath} already exists with entry: ${entry}.`)
   } else {
     appendFileSync(rcPath, entry + '\n')
     console.log(`@lavamoat/allow-scripts: added entry to ${rcPath}: ${entry}.`)
   }
 }
 
+let binsBlockedMemo;
+/**
+ * 
+ * @param {Object} args 
+ * @param {boolean} noMemoization - turn off memoization, make a fresh lookup
+ * @returns {boolean}
+ */
+function areBinsBlocked({ noMemoization = false } = {}) {
+  if(noMemoization || binsBlockedMemo === undefined){
+    binsBlockedMemo = isEntryPresent(NPM.CONF.BINS, NPM.RCFILE) || isEntryPresent(YARN1.CONF.BINS, YARN1.RCFILE)
+    // Once yarn3 support via plugin comes in, this function would need to detect that, or cease to exist.
+  }
+  return binsBlockedMemo
+}
+
 function writeRcFile () {
-  const yarnRcExists = existsSync(addInstallParentDir('.yarnrc'))
-  const yarnYmlExists = existsSync(addInstallParentDir('.yarnrc.yml'))
-  const npmRcExists = existsSync(addInstallParentDir('.npmrc'))
+  const yarnRcExists = existsSync(addInstallParentDir(YARN1.RCFILE))
+  const yarnYmlExists = existsSync(addInstallParentDir(YARN3.RCFILE))
+  const npmRcExists = existsSync(addInstallParentDir(NPM.RCFILE))
   const yarnLockExists = existsSync(addInstallParentDir('yarn.lock'))
 
   const configs = []
   if (yarnRcExists || yarnLockExists) {
     configs.push({
-      file: ".yarnrc",
+      file: YARN1.RCFILE,
       exists: yarnRcExists,
-      entry: "ignore-scripts true",
+      entry: YARN1.CONF.SCRIPTS,
     })
+    if(FT.bins) {
+      configs.push({
+        file: YARN1.RCFILE,
+        exists: yarnRcExists,
+        entry: YARN1.CONF.BINS,
+      })
+    }
   }
   if (yarnYmlExists || yarnLockExists) {
     configs.push({
-      file: ".yarnrc.yml",
+      file: YARN3.RCFILE,
       exists: yarnYmlExists,
-      entry: "enableScripts: false",
+      entry: YARN3.CONF.SCRIPTS,
     })
   }
   if (configs.length === 0) {
     // default to npm, because that's what everyone has anyway
     configs.push({
-      file: ".npmrc",
+      file: NPM.RCFILE,
       exists: npmRcExists,
-      entry: "ignore-scripts=true",
+      entry: NPM.CONF.SCRIPTS,
     })
+    if(FT.bins) {
+      configs.push({
+        file: NPM.RCFILE,
+        exists: npmRcExists,
+        entry: NPM.CONF.BINS,
+      })
+    }
   }
 
   configs.forEach(writeRcFileContent)
 }
 
-function addPreinstallAFDependency () {
+function editPackageJson () {
```

Comment:
I note some overlap and differences with new implementation for https://github.com/LavaMoat/LavaMoat/issues/387 so I guess this might see some structural changes soon again?

(Still seems good to not overthink it, to consider this in isolation, and take that in the following changesets as it becomes necessary, which I guess is the intention here?)

https://github.com/LavaMoat/LavaMoat/compare/main...lava-cli#diff-b64c06b15a9e440c63fed29404849b37a115750c524c7a632e160681a0392290R77

Reply:
This change happened first, but I stopped myself from further structural changes to limit the conflict later ;) 

Once this lands, I'll rebase lava-cli and arrange for this added functionality to use the tooling over there.